astropy/io/ascii/src/tokenizer.c

```diff
@@ -601,10 +601,15 @@ double str_to_double(tokenizer_t *self, char *str)
     {
         val = strtod(str, &tmp);
 
-        if (tmp == str || *tmp != 0)
+        if (tmp == str || *tmp != 0) {
+		    if (0 == strncmp(str, "nan", 3)) {
+				return NAN;
+		    }
```

Comment:
Turns out there are quite a number of cases that need to be supported in principle: http://www.gnu.org/software/libc/manual/html_node/Parsing-of-Floats.html  Do we really care about the "nan(arbitrary characters)" case?  I have never seen this used.  Supporting "inf/infinity" and +/- are straightforward and obvious.